core/src/main/java/com/amplitude/core/Amplitude.kt

```diff
@@ -315,17 +315,32 @@ open class Amplitude internal constructor(
         return this
     }
 
-    private fun process(event: BaseEvent) {
+    protected fun process(event: BaseEvent) {
         if (configuration.optOut) {
             logger.info("Skip event for opt out config.")
             return
         }
+
+        val sessionEvents = processEvent(event)
```

Comment:
Maybe not call it sessionEvents? just events are fine. This way seems to be coupling the logic in core and android package.